hphp/tools/tc-print/tc-print.cpp

```diff
@@ -334,13 +360,13 @@ void loadPerfEvents() {
   char   eventCaption[MAX_SYM_LEN];
   char   line[2*MAX_SYM_LEN];
   TCA    addr;
-  uint32_t tcSamples[NUM_EVENT_TYPES];
-  uint32_t hhvmSamples[NUM_EVENT_TYPES];
+  uint32_t tcSamples[getNumEventTypes()];
```

Comment:
Variable-length arrays are dangerous so we don't use them. Can you rewrite to get rid of them all (eg, use std::vector)?